Deployment: Dockerfile and Smithery config

calclavia opened the pull request:
This PR adds files for deploying your MCP server on Smithery. After deployment, users can use your server over WebSockets (hosted on [Smithery](https://smithery.ai)) without needing to install any dependencies.

### Changes
- Added **Dockerfile** to package your server for deployment.
- Added **Smithery Configuration** file to specify how to start your server. See [documentation](https://smithery.ai/docs/deployments).
- Updated **README** to include installation command via Smithery and a popularity badge. _Note that the command only works after the server is deployed on Smithery._

### Server Details
- **Smithery ID**: `@calebl/ynab-mcp-server`
- **Server Homepage**: [https://smithery.ai/server/@calebl/ynab-mcp-server](https://smithery.ai/server/@calebl/ynab-mcp-server?utm_campaign=pr&modal=claim)

### Action Items
- [X] **Build Passing**: We verified that the Docker builds and your server starts up using an automated test.
- [ ] **Code Review**: Please review the changes to ensure the configuration is accurate for your server.
- [ ] **Claim Server**: Head to your [server page](https://smithery.ai/server/@calebl/ynab-mcp-server?utm_campaign=pr&modal=claim) to claim your server. This will let you edit your server listing on Smithery and deploy new versions of your server.

calebl commented:
@calclavia can you update? I've removed the knowledge environment variables. I decided against storing anything locally. I've added an optional YNAB_BUDGET_ID variable as well

calclavia commented:
> @calclavia can you update? I've removed the knowledge environment variables. I decided against storing anything locally. I've added an optional YNAB_BUDGET_ID variable as well

Feel free to directly update this PR, you should have permission to do so.
